components/team/MemberEditor.tsx

```diff
@@ -0,0 +1,209 @@
+import { ToolsOzoneTeamDefs } from '@atproto/api'
+import { FormEvent, useState } from 'react'
+import { toast } from 'react-toastify'
+
+import { Alert } from '@/common/Alert'
+import { ActionButton } from '@/common/buttons'
+import { Card } from '@/common/Card'
+import { Checkbox, FormLabel, Input, Select } from '@/common/forms'
+import client from '@/lib/client'
+import { getDidFromHandle } from '@/lib/identity'
+import { queryClient } from 'components/QueryClient'
+import { MemberRoleNames } from './Role'
+
+const useMemberEditor = ({
+  isNewMember,
+  onSuccess,
+}: {
+  isNewMember: boolean
+  onSuccess: () => void
+}) => {
+  const [submission, setSubmission] = useState<{
+    isSubmitting: boolean
+    error: string
+  }>({ isSubmitting: false, error: '' })
+
+  const onFormSubmit = async (
+    ev: FormEvent<HTMLFormElement> & { target: HTMLFormElement },
+  ) => {
+    ev.preventDefault()
+    try {
+      setSubmission({ isSubmitting: true, error: '' })
+      const formData = new FormData(ev.currentTarget)
+      let didOrHandle = formData.get(
+        isNewMember ? 'identifier' : 'did',
+      ) as string
+      const role = formData.get('role') as ToolsOzoneTeamDefs.Member['role']
+      let did = didOrHandle
+
+      if (didOrHandle.startsWith('@') || didOrHandle.includes('.')) {
+        const didFromHandle = await getDidFromHandle(didOrHandle)
+        if (!didFromHandle) {
+          setSubmission({
+            isSubmitting: false,
+            error: `Could not resolve handle ${didOrHandle}`,
+          })
+          return
+        }
+        did = didFromHandle
+      }
+
+      // Normally we wouldn't use <any> but the result of the request does not change
+      // anything in the UI so we don't need to type it
+      let request: Promise<any>
```

Comment:
The `unknown` type seems like it may be suitable here, and safer/friendlier from a typing perspective.
```suggestion
      // The result of the request does not change anything in the UI so the type can be unknown
      let request: Promise<unknown>
```